app/helpers/StringHelper.php

```diff
@@ -131,7 +136,14 @@ function getHashtags($string): array
  */
 function echapJid($jid): string
 {
-    return str_replace(' ', '\40', $jid);
+    $jid_parts = explodeJid($jid);
+
+    $from = array('\\',   ' ',    '"',    '&',    '\'',   '/',    ':',    '<',    '>',    '@');
```

Comment:
Looks great but I'll have to check if it doesn't break stuff in the UI

Reply:
When displaying a JID, you SHOULD translate all the characters back, says the standard.